.github/workflows/build.yaml

```diff
@@ -0,0 +1,53 @@
+name: Build
+on:
+    - push
+    - pull_request
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_ACTION: true
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          # TODO: this should not be hardcoded
+          go-version: '1.24.2'
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          driver-opts: network=host
+
+      - name: Set up Docker layer cache
+        uses: actions/cache@v4
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+
+      - name: Install wget
+        run: sudo apt-get update && sudo apt-get install -y wget
+
+      - name: Build image
+        run: |
+          make
+
+      - name: List build outputs
+        run: ls -lh
+
+      - name: Upload artifacts
+        # Allows to test the build locally with act without uploading artifacts
+        if: ${{ env.ACT != 'true' && success() }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: distro-build
+          path: |
+            distro.qcow2
+            distro.tar.xz
+          if-no-files-found: ignore
```

Comment:
Shouldn't we find at least one of the two files?

Reply:
That seems likely to lead us to not noticing that we don't actually have any artifacts…